spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-cloudfoundry/src/main/java/org/springframework/cloud/data/module/deployer/cloudfoundry/CloudFoundryModuleDeploymentIdConverter.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.module.deployer.cloudfoundry;
+
+import org.springframework.cloud.data.core.ModuleDeploymentId;
+import org.springframework.stereotype.Component;
+
+@Component
+public final class CloudFoundryModuleDeploymentIdConverter {
+
+	private static final String PREFIX = "spring_cloud_data_";
+
+	String toApplicationName(ModuleDeploymentId moduleDeploymentId) {
+		return PREFIX + moduleDeploymentId.toString();
+	}
+
+	ModuleDeploymentId toModuleDeploymentId(String applicationName) {
+		try {
+			return ModuleDeploymentId.parse(moduleIdString(applicationName));
+		}
+		catch (NullPointerException e) {
```

Comment:
Is there a null check we can add somewhere instead of checking for an NPE?  Using Exceptions to control logic flow isn't recommended.